project_stage_warning/models/project_task.py

```diff
@@ -0,0 +1,24 @@
+# -- coding: utf-8 --
+from odoo import models, fields, api, _
+from odoo.exceptions import UserError
+
+
+class ProjectTask(models.Model):
+    _inherit = "project.task"
+
+    def write(self, values):
+        if values.get('stage_id'):
+            new_stage = self.env['project.task.type'].sudo().browse(values.get('stage_id'))
+            end_sequence = new_stage.sequence
+            for task in self:
+                start_sequence = task.stage_id.sequence
+                skipped_stage = self.env['project.task.type'].sudo().search([
+                    ('sequence', '<', end_sequence),
+                    ('sequence', '>', start_sequence),
+                    ('project_ids', '=', task.project_id.id),
```

Comment:
task.project_id.id

we might need to check if project is updated on task or not

```suggestion
                    ('project_ids', '=', values.get('project_id', task.project_id.id)),

```

Reply:
Good catch, i've updated it as you suggested